autonomi/src/client/high_level/files/fs_shared.rs

```diff
@@ -99,15 +101,48 @@ impl Client {
 
         let total_chunks = aggregated_chunks.len();
 
+        let mut processed_chunks = 0;
+        let mut retry_on_failure = self.retry_failed != 0;
+
         // Process all chunks for this file in batches
         while !aggregated_chunks.is_empty() {
-            self.process_chunk_batch(
-                &mut aggregated_chunks,
-                &mut receipts,
-                &mut free_chunks_counts,
-                payment_option.clone(),
-            )
-            .await?;
+            let candidate_chunks = aggregated_chunks.len();
+
+            let batch_result = self
+                .process_chunk_batch(
+                    &mut aggregated_chunks,
+                    &mut receipts,
+                    &mut free_chunks_counts,
+                    payment_option.clone(),
+                    retry_on_failure,
+                )
```

Comment:
Not introduced in this PR but significantly more affected with it: is there any way we can remove all these "mut"? They make it hard to know what's going on.